### PR TITLE
Added tests for visible ranges(scroll, fold)

### LINKED ISSUE #160: Test a command's effect on visible ranges
Currently we're storing visible ranges during test case recording, but we are not asserting anything about them. 

This will be interesting because visible ranges depend on the size of the editor, and because the tests will run headlessly in CI.

### To do

- [x] Test folding, esp #401 
- [x] Test scrolling as described in https://github.com/pokey/cursorless-vscode/issues/160#issuecomment-1004807911

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
fixes #160
## Checklist

- [x] I have added [tests](https://github.com/pokey/cursorless-vscode/blob/main/docs/contributing/test-case-recorder.md)

### CODE REVIEW

File: src/test/suite/fold.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    "fold made",
+    "fold",
+    [
+      {
+        type: "primitive",
+        mark: {
+          type: "decoratedSymbol",
+          symbolColor: "default",
+          character: "m",
+        },
```
Comment (pokey) [inline src/test/suite/fold.test.ts:35]: The problem with using a mark here is that this test will break if we change our hat allocation algorithm, whereas for the recorded tests we capture the required allocations.  Maybe the easiest thing would be to just change this test to use the current selection, like we do for some of our [other manual tests](https://github.com/cursorless-dev/cursorless-vscode/blob/30bc0f2ad9ed0d070a256965913afb1e1a675557/src/test/suite/backwardCompatibility.test.ts#L23)?

File: src/test/suite/scroll.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "crown whale",
+    "scrollToTop",
+    [
+      {
+        type: "primitive",
+        mark: {
+          type: "decoratedSymbol",
+          symbolColor: "default",
+          character: "w",
+        },
```
Comment (pokey) [inline src/test/suite/scroll.test.ts:35]: as above, maybe we should just use cursor mark to avoid hat allocation fragility?

**Review by pokey — DISMISSED** [review]
This looks really nice; thanks for following up on this stuff!  I left one minor comment.  Not a showstopper, as we have the same problem in a couple other manual tests, but seems easy to fix and might save us some headaches down the road 🤷‍♂️

But yeah re your comment on the issue, I think it's fine to handle these as one-offs rather than adding support to test case recorder

File: src/test/suite/fold.test.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply src/test/suite/fold.test.ts]: done

**Review by pokey — APPROVED** [review]
Nice 😊

**Pull request merged**